### PR TITLE
Fetch record only once from storage

### PR DESCRIPTION
**Pull request by leplatrem**
This allowed me to go from 29 RPS to 35 RPS (+18%)

@ametaireau r?

### CODE REVIEW

**Comment by ametaireau** [comment]
We could add a unittest on the authorization policy which check that `current_record` is set correctly and another one on resource which checks that the record used is the one from the context.

Otherwise, looks good to me.

File: cliquet/authorization.py [hunk 1]
```diff
@@ -105,11 +106,12 @@ def __init__(self, request):
...
                 # exists, return "write" if it does, "create" otherwise.
 
-                # If the view exists, call it with the request and catch an
+                # If the view exists, use its collection to catch an
                 # eventual NotFound.
                 resource = service.resource(request=request, context=self)
                 try:
-                    resource.collection.get_record(resource.record_id)
+                    record = resource.collection.get_record(resource.record_id)
+                    current_record = record
```
Comment (Natim) [inline cliquet/authorization.py]: can't we just write into current_record directly ?

Comment (leplatrem) [inline cliquet/authorization.py]: just a matter of taste+linewrap :(

**Comment by leplatrem** [comment]
Thanks for your feedback. 2 tests added.

**Pull request merged**